commercetools/utils.go

```diff
@@ -10,13 +10,19 @@ import (
 	mapstructure "github.com/mitchellh/mapstructure"
 )
 
+var (
+	reSlice = regexp.MustCompile(`^([^\[]+)\[\d+\]$`)
+	reMap   = regexp.MustCompile(`^([^\[]+)\[([^\]]+)]$`)
+)
+
 func serializeQueryParams(v interface{}) url.Values {
-	re := regexp.MustCompile(`([^\]]+)\[([^\]]+)]$`)
 	values, _ := query.Values(v)
 
 	newValues := url.Values{}
 	for key, values := range values {
-		newKey := re.ReplaceAllString(key, "$1.$2")
+		// TODO: this is pretty ugly and should be fixed upstream
+		newKey := reSlice.ReplaceAllString(key, "$1")
```

Comment:
It is kind of unclear to me what the purpose of this line is. Could you elaborate a little on what you are trying to achieve?

Reply:
It's a dirty hack to to turn e.g. `filter.0=x&filter.1=y` into `filter=x&filter=y`. The former version is generated by github.com/appscode/go-querystring/query and does not work with the CT API.

This behavior is controllable in the original version of go-querystring (https://github.com/google/go-querystring), but it doesn't support map's yet. I guess that's why you're using the fork...

Reply:
I guess fixing it in github.com/google/go-querystring would be the best approach.

https://github.com/buztard/go-querystring/commit/15c210c17067265eaa335ffc86ad1c1217717824

Reply:
Alright thanks for elaborating. As we are regenerating soon with a new codegen tool (as discussed [here](https://github.com/labd/commercetools-go-sdk/pull/63#issuecomment-818827221)) I am ok with merging for now